xdsl/dialects/llvm.py

```diff
@@ -46,6 +48,141 @@ def parse_parameters(parser: BaseParser) -> list[Attribute]:
+        if isinstance(self.type, NoneAttr):
+            return
+
+        printer.print_string("<")
+        printer.print_attribute(self.type)
+        if not isinstance(self.addr_space, NoneAttr):
+            printer.print_string(", ")
+            printer.print_attribute(self.addr_space)
+
+        printer.print_string(">")
+
+    @staticmethod
+    def parse_parameters(parser: BaseParser) -> list[Attribute]:
+        if not parser.tokenizer.starts_with('<'):
+            return [NoneAttr(), NoneAttr()]
+        parser.parse_characters('<', "llvm.ptr parameters expected")
+        type = parser.try_parse_type()
+        if type is None:
+            parser.raise_error(
+                "Expected first parameter of llvm.ptr to be a type!")
+        if not parser.tokenizer.starts_with(','):
+            parser.parse_characters('>',
+                                    "End of llvm.ptr parameters expected!")
+            return [type, NoneAttr()]
+        parser.parse_characters(',', "llvm.ptr args must be separated by `,`")
+        addr_space = parser.parse_int_literal()
+        parser.parse_characters('>', "End of llvm.ptr parameters expected!")
+        return [type, IntegerAttr.from_params(addr_space, IndexType())]
+
+    @staticmethod
+    def untyped():
+        return LLVMPointerType([NoneAttr(), NoneAttr()])
+
+    @staticmethod
+    def typed(type: Attribute):
+        return LLVMPointerType([type, NoneAttr()])
+
+
+@irdl_op_definition
+class AllocaOp(Operation):
+    name = "llvm.alloca"
+
+    size: Annotated[Operand, IntegerType]
+
+    alignment: OpAttr[AnyIntegerAttr]
+
+    res: OpResult
+
+    @classmethod
+    def get(cls,
+            size: SSAValue | Operation,
+            elem_type: Attribute,
+            alignment: int = 32,
+            as_untyped_ptr: bool = False):
+        attrs: dict[str, Attribute] = {
+            'alignment': IntegerAttr.from_int_and_width(alignment, 64)
+        }
+        if as_untyped_ptr:
+            ptr_type = LLVMPointerType.untyped()
+            attrs['elem_type'] = elem_type
+        else:
+            ptr_type = LLVMPointerType.typed(elem_type)
+
+        return cls.build(operands=[size],
+                         attributes=attrs,
+                         result_types=[ptr_type])
+
+
+@irdl_op_definition
+class IntToPtrOp(Operation):
+    name = "llvm.inttoptr"
+
+    input: Annotated[Operand, IntegerType]
+
+    output: Annotated[OpResult, LLVMPointerType]
+
+    @classmethod
+    def get(cls,
+            input: SSAValue | Operation,
+            ptr_type: Attribute | None = None):
+        if ptr_type is None:
+            ptr_type = LLVMPointerType.untyped()
+        else:
+            ptr_type = LLVMPointerType.typed(ptr_type)
+        return cls.build(operands=[input], result_types=[ptr_type])
+
+
+@irdl_op_definition
+class LoadOp(Operation):
+    name = "llvm.load"
+
+    ptr: Annotated[Operand, LLVMPointerType]
+
+    dereferenced_value: OpResult
```

Comment:
Shouldn't this be `Annotated[OpResult, AnyAttr()]` here?

Reply:
That's the same thing.
`OpResult` just state that this is a result, and the `Annotated` gives constraints on the result type. If none are given, there are no constraints ;)

Reply:
Sure, but it seems to me that we did it with thr annotated in the redt of the repo. What is our prefered style of writing this? Note that people will look at these dialects as examples, so we should make a conscioud decision here IMO.

Reply:
I'd vote for this style, and would be happy to do a pass to homogenise